wrappers/core18.go

```diff
@@ -273,9 +273,13 @@ func AddSnapdSnapServices(s *snap.Info, inter interacter) error {
 			}
 		}
 	}
+
 	// and finally start snapd.service (it will stop by itself and gets
 	// started by systemd then)
-	if err := sysd.Start("snapd.service"); err != nil {
+	// Because of the file lock held on the snapstate by the Overlord, the new
+	// snapd will block there until we release it. For this reaso, we cannot
+	// start the unit in blocking mode.
+	if err := sysd.StartNoBlock("snapd.service"); err != nil {
```

Comment:
could you add a TODO?

//  TODO: move/share this responsibility with daemon so that we can make the start blocking again